pkg/certs/ensure.go

```
@@ -33,7 +37,57 @@ func EnsureCerts(
 	secretName := vClusterName + "-certs"
 	secret, err := currentNamespaceClient.CoreV1().Secrets(currentNamespace).Get(ctx, secretName, metav1.GetOptions{})
 	if err == nil {
-		return downloadCertsFromSecret(secret, certificateDir)
+		err = downloadCertsFromSecret(secret, certificateDir)
+		if err != nil {
+			return err
+		}
+		shouldUpdate, err := updateSecretStruct(secret)
+		if err != nil {
+			return err
+		}
+
+		klog.Info("checking if we need to update the certs")
+		if !shouldUpdate {
+			return nil
+		}
+
+		klog.Info("removing outdated certs")
+		// delete the certs and recreate them
+		cfg, err := createCfg(serviceCIDR, vClusterName, certificateDir, clusterDomain, etcdSans)
+		if err != nil {
+			return err
+		}
+		err = os.Remove(filepath.Join(certificateDir, "apiserver.crt"))
+		if err != nil {
+			klog.Warning("error while removing apiserver.crt: ", err)
+		}
+		err = os.Remove(filepath.Join(certificateDir, "apiserver.key"))
+		if err != nil {
+			klog.Warning("error while removing apiserver.key")
```

this should return an error here